spec/abci/abci++_app_requirements.md

```diff
@@ -23,15 +23,18 @@ returns via `ResponsePrepareProposal` to CometBFT, also known as the prepared pr
 
 Process *p*'s prepared proposal can differ in two different rounds where *p* is the proposer.
 
-* Requirement 1 [`PrepareProposal`, timeliness]: If *p*'s Application fully executes prepared blocks in
-  `PrepareProposal` and the network is in a synchronous period while processes *p* and *q* are in *r<sub>p</sub>*,
+* Requirement 1 [`PrepareProposal`, timeliness]: If the network is in a synchronous period while processes 
+  *p* and *q* are in *r<sub>p</sub>*,
   then the value of *TimeoutPropose* at *q* must be such that *q*'s propose timer does not time out
   (which would result in *q* prevoting `nil` in *r<sub>p</sub>*).
 
-Full execution of blocks at `PrepareProposal` time stands on CometBFT's critical path. Thus,
+`PrepareProposal` stands on CometBFT's critical path. Thus,
 Requirement 1 ensures the Application or operator will set a value for `TimeoutPropose` such that the time it takes
-to fully execute blocks in `PrepareProposal` does not interfere with CometBFT's propose timer.
-Note that violation of Requirement 1 may just lead to further rounds, but will not compromise liveness.
+to fully execute `PrepareProposal` does not interfere with CometBFT's propose timer.
+Note that violation of Requirement 1 may just lead to further rounds, but will not compromise correctness.
```

Comment:
I'd suggest `safety` instead of correctness, then, as using the wrong timeout cannot cause processes to disagree or violate validity. `Liveness`, however, seems incorrect as timeouts can lead to infinitely many extra rounds in which nil is pre-voted.

Reply:
Discussing with @sergio-mena , we reached the following alternative text for **line 34** (GitHub wouldn't let me add it as a suggestion).

```
Note that the violation of Requirement 1 may lead to further rounds, but will not 
compromise liveness because even though `TimeoutPropose` is used as the initial
value for proposal timeouts, CometBFT will be dynamically adjust these timeouts
such that they will eventually be enough for completing `PrepareProposal`.
```